drivers/spi/spi_xmc4xxx.c

```diff
@@ -0,0 +1,366 @@
+/*
+ * Copyright (c) 2022 Schlumberger
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT infineon_xmc4xxx_spi
+
+#define LOG_LEVEL CONFIG_SPI_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(spi_xmc4xxx);
+
+#include "spi_context.h"
+
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/drivers/spi.h>
+
+#include <xmc_spi.h>
+#include <xmc_usic.h>
+
+struct spi_xmc4xxx_config {
+	XMC_USIC_CH_t *spi;
+	const struct pinctrl_dev_config *pcfg;
+	uint8_t miso_src;
+#if defined(CONFIG_SPI_XMC4XXX_INTERRUPT)
+	void (*irq_config_func)(const struct device *dev);
+#endif
+};
+
+struct spi_xmc4xxx_data {
+	struct spi_context ctx;
+};
+
+static void spi_xmc4xxx_shift_frames(const struct device *dev)
+{
+	struct spi_xmc4xxx_data *data = dev->data;
+	const struct spi_xmc4xxx_config *config = dev->config;
+	struct spi_context *ctx = &data->ctx;
+	uint8_t tx_data = 0;
+	uint8_t rx_data;
+	uint32_t status;
+
+	if (spi_context_tx_buf_on(ctx)) {
+		tx_data = ctx->tx_buf[0];
+	}
+
+	XMC_SPI_CH_ClearStatusFlag(config->spi,
+				   XMC_SPI_CH_STATUS_FLAG_TRANSMIT_SHIFT_INDICATION |
+				   XMC_SPI_CH_STATUS_FLAG_RECEIVE_INDICATION |
+				   XMC_SPI_CH_STATUS_FLAG_ALTERNATIVE_RECEIVE_INDICATION);
+
+	XMC_SPI_CH_Transmit(config->spi, tx_data, XMC_SPI_CH_MODE_STANDARD);
+
+	spi_context_update_tx(ctx, 1, 1);
+
+#if defined(CONFIG_SPI_XMC4XXX_INTERRUPT)
+	return;
+#endif
+
+	/* Wait to finish transmitting */
+	while (1) {
```

Comment:
Is there anyway to do this other than an infinite loop? Can we setup an RTOS event or semaphore to let us know when the transmission is complete?

Same comment would apply to the receiving loop on line 69.

Reply:
@ifyall, yes, the interrupt driven version returns in line 57 without ever reaching this code. The while (1) {...} loops are only used by the polling version.

Reply:
Thanks. Given that, I have approve.

Reply:
> Thanks. Given that, I have approve.

Thanks @ifyall , could you Ack on the PR please